microcosm_flask/conventions/crud.py

```diff
@@ -54,9 +54,9 @@ def configure_search(self, ns, definition):
         @qs(definition.request_schema)
         @response(paginated_list_schema)
         def search(**path_data):
-            page = self.page_cls.from_query_string(definition.request_schema)
-            result = definition.func(**merge_data(path_data, page.to_dict(func=identity)))
-            response_data, headers = page.to_paginated_list(result, ns, Operation.Search)
```

Comment:
Here we create the response page based on the same `page` as we use for actual computing. If the schema contain `QueryStringList` (or sth similar) the deserialized value in the `page` is of type `List` so it got serialized by https://github.com/pawmarkor/microcosm-flask/blob/3aeda476d8c2f0cc1e615a136d778c9a83715d24/microcosm_flask/paging.py#L174 into a list based string i.e. `[a,b,c]` so `QueryStringList` is not used for serialization at all in this case and we got a wrong query string in the response link.